lib/stop.go

```diff
@@ -24,6 +24,9 @@ func (c *ContainerServer) ContainerStop(ctx context.Context, container string, t
 			if err := c.runtime.StopContainer(ctx, ctr, timeout); err != nil {
 				return "", errors.Wrapf(err, "failed to stop container %s", ctrID)
 			}
+			if err := oci.WaitContainerStateStopped(ctx, ctr, timeout); err != nil {
```

Comment:
We can add a state check here too for the runc case to avoid spawning goroutine.

Reply:
We can do that from the function itself then. We won't spawn the internal go routine with the for loop if we are already in the expected state, right ?

Reply:
Sure, that works too.

Reply:
done